pkg/scalers/metrics_api_scaler.go

```diff
@@ -0,0 +1,155 @@
+package scalers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	kedautil "github.com/kedacore/keda/pkg/util"
+	"io/ioutil"
+	"k8s.io/api/autoscaling/v2beta2"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	"net/http"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"strconv"
+	"strings"
+)
+
+type metricsAPIScaler struct {
+	metadata *metricsAPIScalerMetadata
+}
+
+type metricsAPIScalerMetadata struct {
+	targetValue int
+	url         string
+	metricName  string
+}
+
+type metric struct {
+	Name  string  `json:"name"`
+	Value float64 `json:"value"`
+}
+
+var httpLog = logf.Log.WithName("metrics_api_scaler")
+
+// NewMetricsAPIScaler creates a new HTTP scaler
+func NewMetricsAPIScaler(resolvedEnv, metadata, authParams map[string]string) (Scaler, error) {
+	meta, err := metricsAPIMetadata(resolvedEnv, metadata, authParams)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing metric API metadata: %s", err)
+	}
+	scaler := &metricsAPIScaler{metadata: meta}
+	err = scaler.checkHealth()
```

Comment:
FYI, the NewMetricsAPIScaler is not called only when the scaler is created (ie. when you create a ScaledObject). But currently it is called periodically. So the `scaler.checkHealth()` is happening everytime.

Reply:
Got it, but still, I think it made sense to check API health at this moment to avoid "run time" error when checking the metric